worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -30,28 +60,136 @@ public Job putJobInProgress(String jobId) {
         final Job job = jobRepository.findByJobUuid(jobId);
         Assert.notNull(job, String.format("Job %s not found", jobId));
 
-
         // validate status is SUBMITTED
         if (!SUBMITTED.equals(job.getStatus())) {
-            final String errMsg = String.format("Job %s is not in %s status. Skipping job", jobId, SUBMITTED);
+            final String errMsg = String.format("Job %s is not in %s status.", jobId, SUBMITTED);
             throw new IllegalArgumentException(errMsg);
         }
 
         job.setStatus(IN_PROGRESS);
 
-        log.info("Job [{}] is now IN_PROGRESS", job.getId());
         return jobRepository.save(job);
     }
 
     @Override
-    public void completeJob(Job job) {
+    public Job processJob(final String jobId) {
+
+        final Job job = jobRepository.findByJobUuid(jobId);
+        log.info("Found job : {} - {} - {} ", job.getId(), job.getJobUuid(), job.getStatus());
+
+        final Sponsor sponsor = job.getUser().getSponsor();
+
+        final List<Contract> attestedContracts = sponsor.getAttestedContracts();
+        log.info("number of attested contracts : {}", attestedContracts.size());
+
+
+        final Path outputDirPath = Paths.get(efsMount, job.getJobUuid());
+        final Path outputDir = fileService.createDirectory(outputDirPath);
+        for (Contract contract : attestedContracts) {
+            log.info("contract : {} ", contract.getContractNumber());
+
+            JobOutput jobOutput = null;
+            try {
+                jobOutput = processContract(outputDir, contract);
+                jobOutput.setError(false);
+            } catch (Exception e) {
+                jobOutput.setError(true);
+                log.error("error processing contract : {} ", contract.getContractNumber(), e);
+            }
+
+            job.addJobOutput(jobOutput);
+            jobOutputRepository.save(jobOutput);
+        }
+
+        completeJob(job);
+        return job;
+    }
+
+
+    private JobOutput processContract(final Path outputDir, Contract contract) {
+        final String filename = contract.getContractNumber() + NDJSON_EXTENSION;
+        final var ndJsonFile = fileService.createFile(outputDir, filename);
+
+        final var patientsByContract = beneficiaryAdapter.getPatientsByContract(contract.getContractNumber());
+
+        final var futureResourcesHandles =  new ArrayList<Future<List<Resource>>>();
+
+        int counter = 0;
+        for (var patient : patientsByContract.getPatients()) {
+            final var resources = bfdClientAdapter.getEobBundleResources(patient.getPatientId());
+            futureResourcesHandles.add(resources);
+
+            ++counter;
+            if (counter % GROUP_SIZE == 0) {
+                processResources(futureResourcesHandles, ndJsonFile);
+            }
+        }
+
+        processResources(futureResourcesHandles, ndJsonFile);
+
+        JobOutput jobOutput = new JobOutput();
+        jobOutput.setFilePath(getFilePath(ndJsonFile));
```

Comment:
The file path here is supposed to be relative to the EFS mount, not the absolute one. More portability that way.

Reply:
okay. After change the filepath will look something like this:

```
JOB_11/CONTRACT_S0011.ndjson
```

Reply:
Correct.

Reply:
done